provisioning-telephony/nodejs/index.js

```diff
@@ -0,0 +1,331 @@
+                    instance: inputTemplate.sipTrunk.instance, // BYOC signaling IP
+                },
+            },
+            trunk_protocol: {
+                type: 'string',
+                enum: ['SIP'],
+                value: {
+                    default: 'SIP',
+                    instance: 'SIP',
+                },
+            },
+            trunk_sip_authentication_credentials_realm: {
+                type: 'string',
+                value: {
+                    default: '',
+                    instance: inputTemplate.sipTrunk.realm, // Realm
+                },
+            },
+            trunk_sip_authentication_credentials_username: {
+                type: 'string',
+                value: {
+                    default: '',
+                    instance: inputTemplate.sipTrunk.userName, // User Name
+                },
+            },
+            trunk_sip_authentication_credentials_password: {
+                type: 'string',
+                value: {
+                    default: '',
+                    instance: inputTemplate.sipTrunk.password, // Password
+                },
+            },
+            trunk_outboundIdentity_callingName: {
+                type: 'string',
+                pattern: '^[\\S ]{0,40}$',
+                value: {
+                    default: '',
+                    instance: inputTemplate.sipTrunk.callingName, // Calling Name
+                },
+            },
+            trunk_outboundIdentity_callingName_overrideMethod: {
+                type: 'string',
+                enum: ['Always', 'Unassigned DID'],
+                value: {
+                    default: 'Always',
+                    instance: 'Always',
+                },
+            },
+            trunk_outboundIdentity_callingAddress: {
+                type: 'string',
+                value: {
+                    default: '',
+                    instance: inputTemplate.sipTrunk.address, // Calling Address
+                },
+            },
+            trunk_outboundIdentity_callingAddress_overrideMethod: {
+                type: 'string',
+                enum: ['Always', 'Unassigned DID'],
+                value: {
+                    default: 'Always',
+                    instance: 'Always',
+                },
+            },
+            trunk_outboundIdentity_calledAddress_omitPlusPrefix: {
+                type: 'boolean',
+                value: {
+                    default: false,
+                    instance: false,
+                },
+            },
+            trunk_outboundIdentity_callingAddress_omitPlusPrefix: {
+                type: 'boolean',
+                value: {
+                    default: false,
+                    instance: false,
+                },
+            },
+            trunk_sip_termination_uri: {
+                type: 'string',
+                value: {
+                    instance: inputTemplate.sipTrunk.sipServers, // Inbound SIP Termination Identifier
```

Comment:
FYI: This value is globally unique. There is a possibility that two customer may try the same name. For example _mytesttrunk_.

Reply:
Will take note of this in the step by step section.